src/sage/schemes/elliptic_curves/ell_finite_field.py

```diff
@@ -2786,3 +2861,290 @@ def find_q(m, m4_fac, D):
                         yield Et
                     except ValueError:
                         pass
+
+def EllipticCurve_with_prime_order(N):
```

Comment:
```suggestion
def elliptic_curves_with_prime_order(N):
```

and make the corresponding change wherever the function is used.

Unless there is some (well) established naming convention, I don't like the current naming because it sounds like it constructs a single elliptic curve, when in reality it is an iterator of elliptic curves. If some naming convention I'm unaware of necessitates the capital letters for the name, can we at least have `EllipticCurves_with_prime_order`?

Reply:
I will consult opinions from other Sage contributors, but seeing that `EllipticCurve_with_order` got merged in another PR, I don't see any problem with the naming.

Reply:
If there is an established convention here then you don't need to change it. I'd still prefer `EllipticCurves_with_prime_order` (plural curves) though but not a huge deal.